### DISCUSSION TITLE
Can Pacstall work on Arch and Arch based distros?

### DISCUSSION THREAD

**Discussion by trymeouteh** [pacstall/pacstall#552] [Q&A] [answered]
The AUR is great but one downside to the AUR is that it is centralised. Pacstall on the other hand are federated since you can self host your own Pacstall repository and add/remove Pacstall repositories into your Pacstall CLI to download packages from. This is not possible with the AUR since there is only one repository available.

**Comment by wizard-28** [comment, marked as answer]
No, Pacstall is designed to only work for the Debian family. Other distribution families are not supported.

**Reply by D-Brox** [reply]
Technically, if you had `apt` installed and configured, pacstall would work. But then it would be really pure arch, to be honest.

**Comment by Mar0xy** [comment]
> the AUR is that it is centralised

This in fact is not true you can actually self-host an arch repo which in the end would be something similar to AUR as AUR is basically just users sharing their PKGBUILD files to let people make a pkg.tar.xz file on their system. Hosting an arch repo would eliminate this overhead of people having to build the pkgs themself.

see https://wiki.archlinux.org/title/unofficial_user_repositories and https://wiki.archlinux.org/title/Pacman/Tips_and_tricks#Custom_local_repository

there are also custom tools to make this easy https://github.com/kalinda99/ARepoBuilder

but if you do want something similar to AUR you can easily setup your own aurweb version by cloning the git repo and running the dockerfile inside it.

**Reply by trymeouteh** [reply]
I know you can host your own Arch repo but the software from the official Arch repo and any 3rd party repos are compiled software. The AUR is one repository which is essentially a centralised library of PKBUILDs one can index using yay or Pamac. You could host your own PKBUILD repo but there is no way to add it to ones computer and be able to use yay or Pamac to get the PKBUILDs from the 3rd party repo/PKBUILD library and download the source code and compile it.

**Reply by Mar0xy** [reply]
so don't yay and pamac their entire purpose is to use the APIs provided by aurweb to get the PKGBUILDs and etc as Arch doesn't have any linking to AUR built in anyways.
So, in theroy hosting an version of aurweb and then editing yay or pamac to have them download from that instance would work.

**Reply by trymeouteh** [reply]
Do you have to switch back and fourth between repositories or can you just add multiple repos without switching back and fourth?
